mscore/inspector/inspectorBase.cpp

```diff
@@ -48,6 +48,12 @@ InspectorBase::InspectorBase(QWidget* parent)
       _layout->setSpacing(0);
       _layout->setContentsMargins(0, 10, 0, 0);
       _layout->addStretch(100);
+      scrollPreventer = new InspectorScrollPreventer();
+      }
+
+InspectorBase::~InspectorBase()
+      {
+      delete scrollPreventer;
```

Comment:
It would probably be better to make this `InspectorBase` a parent of `InspectorScrollPreventer` or make `scrollPreventer` a plain variable of `InspectorScrollPreventer` type rather than a pointer. Then we wouldn't need to manage its lifetime manually.

Reply:
Sorry, by plain variable, you mean just creating it with `scrollPreventer = InspectorScrollPreventer()` and then making `scrollPreventer` not a pointer? Won't that store it on the stack instead, or am I misunderstanding that?

Reply:
As it will still be a member of `InspectorBase`, it will be stored where this `InspectorBase` instance is stored. It will just not be allocated separately on a heap.

Reply:
Gotcha, thank you!

Reply:
Right, I can't make this change, as in, I don't think it's actually possible. Here's why:

The definition `QObject::installEventFilter(QObject* eventFilter)` takes a pointer. If the eventFilter is a member of `InspectorBase` rather than just referred to by a pointer, then I would need to create a pointer to it when passing it to this function. However, this line:

```cpp
w->installEventFilter(&scrollPreventer);
```

causes an error:

```error: ISO C++ forbids taking the address of an unqualified or parenthesized non-static member function to form a pointer to member function```

It's worth nothing that I also had to change the definition in the header file to:

```cpp
class InspectorBase : public QWidget {
// ...
      InspectorScrollPreventer scrollPreventer(InspectorScrollPreventer());
// ...
}
```

which was the only way I could get the definition to work.

So, I even if there is a way to work around, I don't think it's too much of a risk to manage it as a pointer. It's not like the destructor will go wrong and just forget to delete it :)

Reply:
Actually the simplest way would anyway to pass a parent to `QObject` constructor in when constructing `InspectorScrollPreventer`, but it should be safe to take an address of member variable as well (maybe some name collision with member function happens here?)

Still with the upcoming Inspector redesign this change may happen to be not applicable to the new implementation so it might be better now not to fix the old implementation now. Still if some new release happens to contain the old inspector this PR will definitely be useful.

Reply:
OK, your first way using a parent sounds good. That would mean I would still construct it with `new` and store it as a pointer, right?

Also, who's working on the new inspector design? And, which release are we hoping to see it in?

Reply:
> OK, your first way using a parent sounds good. That would mean I would still construct it with `new` and store it as a pointer, right?

Yes, in this case this will still be a pointer, just with lifetime managed by QObject implementation instead of our own.

> Also, who's working on the new inspector design? And, which release are we hoping to see it in?

@vpereverzev is working on its implementation. As far as I understand, the current plan is to include the new inspector in 3.5 release.

Reply:
Brilliant, thanks.

Reply:
Ok, fixed now. Up to you whether it's worth merging or not :man_shrugging: